instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -355,39 +434,55 @@ def handle_request(
         method, url, headers, stream, extensions = _extract_parameters(
             args, kwargs
         )
-        span_attributes = _prepare_attributes(method, url)
+        method_original = method.decode()
+        span_name = _get_default_span_name(method_original)
+        span_attributes = {}
+        # apply http client response attributes according to semconv
+        _apply_request_client_attributes_to_span(
+            span_attributes, url, method_original, span_name, self.semconv
+        )
 
         request_info = RequestInfo(method, url, headers, stream, extensions)
-        span_name = _get_default_span_name(
-            span_attributes[SpanAttributes.HTTP_METHOD]
-        )
 
         with self._tracer.start_as_current_span(
             span_name, kind=SpanKind.CLIENT, attributes=span_attributes
         ) as span:
-            if self._request_hook is not None:
+            exception = None
+            if callable(self._request_hook):
                 self._request_hook(span, request_info)
 
             _inject_propagation_headers(headers, args, kwargs)
-            response = self._transport.handle_request(*args, **kwargs)
-            if isinstance(response, httpx.Response):
-                response: httpx.Response = response
-                status_code = response.status_code
-                headers = response.headers
-                stream = response.stream
-                extensions = response.extensions
-            else:
-                status_code, headers, stream, extensions = response
-
-            _apply_status_code(span, status_code)
-
-            if self._response_hook is not None:
-                self._response_hook(
-                    span,
-                    request_info,
-                    ResponseInfo(status_code, headers, stream, extensions),
+
+            try:
+                response = self._transport.handle_request(*args, **kwargs)
+            except Exception as exc:  # pylint: disable=W0703
+                exception = exc
+                response = getattr(exc, "response", None)
+
+            if isinstance(response, (httpx.Response, tuple)):
+                status_code, headers, stream, extensions, http_version = (
+                    _extract_response(response)
                 )
 
+                if span.is_recording():
+                    # apply http client response attributes according to semconv
+                    _apply_response_client_attributes_to_span(
+                        span, status_code, http_version, self.semconv
+                    )
+
+                if callable(self._response_hook):
+                    self._response_hook(
+                        span,
+                        request_info,
+                        ResponseInfo(status_code, headers, stream, extensions),
+                    )
+
+            if exception and _report_new(self.semconv):
```

Comment:
Think this should be within the `is_recording()` check.

Reply:
Edit: Can probably handle setting `error.type` to exception name if you use [set_status](_apply_response_client_attributes_to_span) and pass in `status_code_str`

Reply:
Do you mean something like that:

```
if span.is_recording() and exception is not None:
  _set_status(span, {}, type(exception).__qualname__, -1, self._sem_conv_opt_in_mode)
```

Reply:
I think the `if exception and _report_new(self._sem_conv_opt_in_mode):` makes it more explicit that will apply only for new semconv no?

Reply:
We don't set attributes if the span is not recording.

Reply:
@lzchen, I agree. I was rreferring to _set_status. I made an update to check if it is recording and created a test to validate this.